### PR TITLE
Fix Turreted units not dropping cancelled targets.

### LINKED ISSUE #11027: "Stop" is invalid for defensive structures
Cannot stop any defensive structure from force attacking. I tried pressing "S", but it didn't work. It only works for infantries and vehicles.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #11027.

I don't see a cleaner way to work around this without doing a major rewrite.
The attack code is so horrible :disappointed:

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs [hunk 1]
```diff
@@ -103,7 +104,13 @@ public override Activity Tick(Actor self)
 					var maxRange = targetIsMobile ? new WDist(Math.Max(weapon.Weapon.MinRange.Length, modifiedRange.Length - 1024))
 						: modifiedRange;
 
+					// Check that AttackFollow hasn't cancelled the target by modifying attack.Target
+					// Having both this and AttackFollow modify that field is a horrible hack.
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: > this and AttackFollow

Isn't this also `AttackFollow`?

Comment (pchote) [inline OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: No, this is `AttackActivity.AttackActivity`.

**Comment by Mailaender** [comment]
Fixes the problem as advertised. :white_check_mark:

**Comment by obrakmann** [comment]
:+1: works and doesn't impact the original bug fix. The gunboat in gdi01 continues to work as well.

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/abfdd5a44d3aa17631772323bfb7290eacd7f5f5)

**Pull request merged**